hub/config/src/test/java/uk/gov/ida/hub/config/domain/filters/IdpPredicateFactoryPredicatesTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.ida.hub.config.domain.filters;
+
+import com.google.common.base.Predicate;
+import org.junit.Test;
+import uk.gov.ida.hub.config.domain.IdentityProviderConfigEntityData;
+import uk.gov.ida.hub.config.domain.LevelOfAssurance;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ida.hub.config.domain.filters.PredicateTestHelper.*;
+
+public class IdpPredicateFactoryPredicatesTest {
+
+    private final IdpPredicateFactory idpPredicateFactory = new IdpPredicateFactory();
+
+    @Test
+    public void shouldReturnIdpsForLoaForNonOnboardingTransactionEntity() {
+        final Set<Predicate<IdentityProviderConfigEntityData>> loa1Predicate = idpPredicateFactory
+                .createPredicatesForTransactionEntityAndLoa(transactionEntityNonOnboarding, LevelOfAssurance.LEVEL_1);
+
+        final Set<IdentityProviderConfigEntityData> filteredIdps = getFilteredIdps(allIdps, loa1Predicate);
+
+        final IdentityProviderConfigEntityData[] expectedFilteredIdps = {nonOnboardingLoa1Idp, nonOnboardingAllLevelsIdp,
+                onboardingLoa2Idp, onboardingLoa2IdpOtherOnboardingEntity};
+
+        assertThat(filteredIdps).hasSameSizeAs(expectedFilteredIdps);
```

Comment:
hasSameSizeAs should be unnecessary when you are asserting containsOnly.

From docs for containsOnly:
`Verifies that the actual group contains only the given values and nothing else`

Reply:
Just me being extra pedantic.... The extra check can be removed if it's confusing but I don't think it hurts to have it there.